extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java

```diff
@@ -195,7 +195,20 @@
             throw e;
         }
     }
-    
+
+    private HttpServer getHttpServer(int port, HttpServerProcessor serverProcessor) {
+        ServiceLoader<HttpServer> httpServers = ServiceLoader.load(HttpServer.class);
+        for (HttpServer server : httpServers) {
+            Class<? extends HttpServer> aClass = server.getClass();
+            try {
+                return aClass.getConstructor(int.class, HttpServerProcessor.class).newInstance(port, serverProcessor);
+            } catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException ignored) {
```

Comment:
Please use a java.util Logger to show the reason why it failed ... use log level WARNING